src/workerd/jsg/fast-api.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+
+#include <workerd/jsg/web-idl.h>
+
+#include <v8-fast-api-calls.h>
+#include <v8-local-handle.h>
+#include <v8-value.h>
+
+#include <kj/common.h>
+
+#include <type_traits>
+#include <utility>
+
+namespace workerd::jsg {
+
+// these types are passed by fast api as is and do not require wrapping/unwrapping.
+template <typename T>
+constexpr bool isFastApiPrimitiveType =
+    kj::isSameType<T, void>() || kj::isSameType<T, bool>() || kj::isSameType<T, int32_t>() ||
+    kj::isSameType<T, uint32_t>() || kj::isSameType<T, float>() || kj::isSameType<T, double>();
+
+template <typename T>
+concept FastApiPrimitive = isFastApiPrimitiveType<T>;
```

Comment:
This feels a bit weird. Shouldn't this be something like...

```cpp
template <typename T>
concept FastApiPrimitive = requires {
  kj::isSameType<T, void>() ||
  kj::isSameType<T, bool>() ||
  kj::isSameType<T, int32_t>() ||
  kj::isSameType<T, uint32_t>() ||
  kj::isSameType<T, double>();
};
```

Or something like that... then you can replace the various other templates with the `FastApiPrimitive` concept...

```cpp
template <typename TypeWrapper, FastApiPrimitive T>
constexpr bool isFastApiWrappedType = isWrappedType<TypeWrapper, T> ;
```

or something like that... with concepts you really ought to be able to streamline this.

Reply:
Altough, I like the idea, if you use "requires" on concepts, it will not be possible to test these with static assertions. This is because concepts work differently than traditional SFINAE and template specialization patterns. This occurs because concepts with requires are evaluated as constraints rather than as substitution failures. static_asserts with a concept is immediately evaluated whereas with template type traits, the assertion might only be evaluated when the template is instantiated.

tagging @danlapid since he asked for more concepts as well.

Reply:
For example, taking this example:

```
template <typename TypeWrapper, FastApiPrimitive T>
constexpr bool isFastApiWrappedType = isWrappedType<TypeWrapper, T> ;
```

The following code will not compile:

```
static_assert(!isFastApiWrappedType<kj::String>);
```